src/openzaak/components/documenten/models.py

```diff
@@ -1292,3 +1308,34 @@ def get_informatieobject(self, permission_main_object=None):
             return getattr(self, permission_main_object).latest_version
         else:
             return self.informatieobject.latest_version
+
+
+class ReservedDocument(models.Model):
+    identificatie = models.CharField(
+        max_length=40,
+        blank=True,
+        default="",
+        help_text="Een binnen een gegeven context ondubbelzinnige referentie "
+        "naar het INFORMATIEOBJECT.",
+        db_index=True,
+    )
+    bronorganisatie = RSINField(
+        max_length=9,
+        help_text="Het RSIN van de Niet-natuurlijk persoon zijnde de "
+        "organisatie die het informatieobject heeft gecreëerd of "
+        "heeft ontvangen en als eerste in een samenwerkingsketen "
+        "heeft vastgelegd.",
+        db_index=True,
+    )
+    date_created = models.DateTimeField(
+        auto_now_add=True,
+        help_text=_("De datum en tijd waarop het document werd gereserveerd."),
+    )
+
+    class Meta:
+        unique_together = ("bronorganisatie", "identificatie")
```

Comment:
`unique_together` might be deprecated in a future release of Django, so it's generally recommended to use `UniqueConstraint` (https://docs.djangoproject.com/en/5.2/ref/models/options/#django.db.models.Options.unique_together)
```suggestion
        constraints = [
            models.UniqueConstraint(
                fields=["bronorganisatie", "identificatie"],
                name="bronorganisatie_identificatie_unique_together",
            ),
        ]
```